frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -32,7 +34,8 @@ export function isArgoWorkflowTemplate(template: Workflow): boolean {
 
 export function isTemplateV2(templateString: string): boolean {
   try {
-    const template = jsyaml.safeLoad(templateString);
+    const template =
+      jsyaml.safeLoad(templateString)[PIPELINE_SPEC] ?? jsyaml.safeLoad(templateString);
```

Comment:
I feel that this line should be abstracted to a function so it can be reused everywhere. (And with a test case to demonstrate how this logic differentiate single yaml file vs multiple yaml files. This piece of behavior can be hard to understand because it relies on an unusual API behavior.